models/transfers/ethereum/erc20/transfers_ethereum_erc20.sql

```diff
@@ -19,19 +19,19 @@ with
     ,
     received_transfers as (
         select
-        'receive' || '-' || evt_tx_hash || '-' || evt_index || '-' || `from` as unique_transfer_id,
+        'receive' || '-' || evt_tx_hash || '-' || CAST(evt_index AS VARCHAR(100)) || '-' || `from` as unique_transfer_id,
         `from` as wallet_address,
         contract_address as token_address,
         evt_block_time,
-        - value as amount_raw
+        '-' || value as amount_raw
```

Comment:
I don't think this change is right @belen-pruvost, value here is a double or bigint

Reply:
`||` is acting like a concatenation, and `amount_raw` is a string. Therefore what this does its to save the negative value as a string. I've tested it on Databricks and Spark

Reply:
Yeah I got that, I'm just saying it doesn't really make sense that this value is a string I guess

Reply:
Ok, let me review this and I'll get back to you with findings or a fix 😅

Reply:
This is a CTE that is then used to define the model for `transfers_ethereum.erc20`. 

Without my changes, currently the `amount_raw` column on `transfers_ethereum.erc20` is a string:

<img width="850" alt="Screenshot 2022-12-07 at 13 55 22" src="https://user-images.githubusercontent.com/15678584/206185373-3c7512f2-9cd9-45b7-bb7f-f029e71931a8.png">

In order to populate the `amount_raw` field, I'm unioning 4 different CTES (`sent_transfers`, `received_transfers`, `deposited_weth` and `withdrawn_weth`).

1) On `sent_transfers` the value is taken from `erc20_ethereum.evt_transfer value`. This field is a string ([source](https://dbc-7ee55e77-eabe.cloud.databricks.com/sql/explore/data/hive_metastore/erc20_ethereum/evt_transfer?o=4190327660257548))

2) On `received_transfers` the value is taken from `erc20_ethereum.evt_transfer value`, converted to negative. This field is a string ([source](https://dbc-7ee55e77-eabe.cloud.databricks.com/sql/explore/data/hive_metastore/erc20_ethereum/evt_transfer?o=4190327660257548)), and concatenated with the `-` sign is also a string

3) On `deposited_weth` the value is taken from `zeroex_ethereum.weth9_evt_deposit wad`. This field is a string ([source](https://dbc-7ee55e77-eabe.cloud.databricks.com/sql/explore/data/hive_metastore/zeroex_ethereum/weth9_evt_deposit?o=4190327660257548))

4) On `withdrawn_weth` the value is taken from `zeroex_ethereum.weth9_evt_deposit  wad`, converted to negative. This field is a string ([source](https://dbc-7ee55e77-eabe.cloud.databricks.com/sql/explore/data/hive_metastore/erc20_ethereum/evt_transfer?o=4190327660257548)), and concatenated with the `-` sign is also a string

So the `amount_raw` field from each of the 4 CTEs is a string, which then ends up being a string on the resulting model of `ransfers_ethereum.erc20`. The fix of my PR aims to fix the string concatenation on Trino.

@soispoke please let me know if I'm missing something or not following correctly your concern

Reply:
Okay, I get your point now. I guess this value should ideally be a bigint, but currently it's a string and I just wanted to keep it as it was to avoid breaking more things 😅